Tuist/ProjectDescriptionHelpers/AppCustomConfiguration.swift

```diff
@@ -0,0 +1,107 @@
+import ProjectDescription
+
+public enum AppCustomConfiguration {
+    case debug, betaDevelopment, betaStage, betaProduction, release
+    
+    internal func customConfiguration(with name: String, projectVersion: Version) -> CustomConfiguration {
+        switch self {
+        case .debug:
+            return CustomConfiguration.debug(name: configurationName, settings: settings(with: name, projectVersion: projectVersion))
+        case .betaDevelopment, .betaProduction, .betaStage, .release:
+            return CustomConfiguration.release(name: configurationName, settings: settings(with: name, projectVersion: projectVersion))
+        }
+    }
+    
+    internal func customTargetConfiguration(with name: String) -> CustomConfiguration {
+        switch self {
+        case .debug:
+            return CustomConfiguration.debug(name: configurationName, settings: targetSettings(with: name))
+        case .betaDevelopment, .betaProduction, .betaStage, .release:
+            return CustomConfiguration.release(name: configurationName, settings: targetSettings(with: name))
+        }
+    }
+    
+    private var configurationName: String {
+        switch self {
+        case .debug:
+            return "Debug"
+        case .betaDevelopment:
+            return "Beta-Development"
+        case .betaStage:
+            return "Beta-Stage"
+        case .betaProduction:
+            return "Beta-Production"
```

Comment:
Please revise also [Environments](https://github.com/AckeeCZ/iOS-MVVM-ProjectTemplate/wiki/Environments) section in wiki. These new configurations are not described there yet, I guess we forgot 🙈

Reply:
it's ready, will push it once this is merged 🙂